### PR TITLE
Use `#[MapRequestPayload]` for admin blog controller

### LINKED ISSUE #1434: Use DTOs to map requests (using MapRequestPayload & MapQueryString)
The demo is a good place to show the best practices. Now that we have the new attributes to map requests to DTOs (`MapRequestPayload` & `MapQueryString`) I think it would be great to use them in the demo. It would also help to see some potential issues we can hit using those (there are some open issues related to that).

If you think we should not, feel free to close this issue.

### PR DESCRIPTION
**Pull request by BafS**
Fix #1434

Unfortunately there is no Rest endpoints in the demo, I think it would have been more interesting to use it then.

### CODE REVIEW

File: src/Controller/Admin/BlogController.php [hunk 1]
```diff
@@ -159,12 +161,9 @@ public function edit(Request $request, Post $post, EntityManagerInterface $entit
      */
     #[Route('/{id}/delete', name: 'admin_post_delete', methods: ['POST'])]
     #[IsGranted('delete', subject: 'post')]
-    public function delete(Request $request, Post $post, EntityManagerInterface $entityManager): Response
+    public function delete(#[MapRequestPayload] DeletePostPayload $deletePostPayload, Post $post, EntityManagerInterface $entityManager): Response
```
Comment (OskarStark) [inline src/Controller/Admin/BlogController.php:164]:
```suggestion
    public function delete(
    #[MapRequestPayload] DeletePostPayload $deletePostPayload,
    Post $post,
    EntityManagerInterface $entityManager,
    ): Response
```

**Comment by BafS** [comment]
@94noni I didn't do it because of https://github.com/symfony/demo/issues/1434#issuecomment-1761098715 "Maybe we could do it only in one controller". Unfortunately there is not much to do in any controller. The best would be to introduce a restful api in the demo.

**Comment by 94noni** [comment]
oki got it
for the api part, there were already a [discuss on that topic](https://github.com/symfony/demo/issues/1344)

File: src/Dto/DeletePostPayload.php [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace App\Dto;
```
Comment (94noni) [inline src/Dto/DeletePostPayload.php:12]: I personally am used to `\Model` namespace
`\Dto`, for a demo, would make sens only if the dev reading know it (what we can imagine it is ok, but not sure :s)
wdyt?

Comment (BafS) [reply src/Dto/DeletePostPayload.php:12]: Model namespace is often used for database models (in Laravel for example), I think having Dto is less confusing

Comment (stof) [reply src/Dto/DeletePostPayload.php:12]: In Doctrine, the database models are in the `Entity` namespace.

**Comment by javiereguiluz** [comment]
Closing as "won't merge" for the reasons explained in https://github.com/symfony/demo/issues/1434#issuecomment-1987942195

Sorry!

**Pull request closed without merging**